dcrsqlite/apisource.go

```diff
@@ -740,3 +675,261 @@ func (db *wiredDB) GetAddressTransactionsRaw(addr string, count int) []*apitypes
+	tx.Time = data.Time
+	t := time.Unix(tx.Time, 0)
+	tx.FormattedTime = t.Format("1/_2/06 15:04:05")
+	tx.Confirmations = data.Confirmations
+	return tx
+}
+
+func (db *wiredDB) GetExploreBlocks(start int, end int) []*explorer.BlockBasic {
+	summaries := make([]*explorer.BlockBasic, 0, start-end)
+	for i := start; i > end; i-- {
+		data := db.GetBlockVerbose(i, false)
+		block := makeExplorerBlockBasic(data)
+		summaries = append(summaries, block)
+	}
+	return summaries
+}
+
+func (db *wiredDB) GetExplorerBlock(hash string) *explorer.BlockInfo {
+	data := db.GetBlockVerboseByHash(hash, true)
+	if data == nil {
+		log.Error("Unable to get block for block hash " + hash)
+	}
+
+	// Explorer Block Basic
+	blockBasic := makeExplorerBlockBasic(data)
+
+	block := &explorer.BlockInfo{
+		BlockBasic: blockBasic,
+	}
+	block.Hash = data.Hash
+	block.Version = data.Version
+	block.Confirmations = data.Confirmations
+	block.StakeRoot = data.StakeRoot
+	block.MerkleRoot = data.MerkleRoot
+	block.Nonce = data.Nonce
+	block.VoteBits = data.VoteBits
+	block.FinalState = data.FinalState
+	block.PoolSize = data.PoolSize
+	block.Bits = data.Bits
+	block.SBits = data.SBits
+	block.Difficulty = data.Difficulty
+	block.ExtraData = data.ExtraData
+	block.StakeVersion = data.StakeVersion
+	block.PreviousHash = data.PreviousHash
+	block.NextHash = data.NextHash
+
+	votes := make([]*explorer.TxBasic, 0, block.Voters)
+	revocations := make([]*explorer.TxBasic, 0, block.Revocations)
+	tickets := make([]*explorer.TxBasic, 0, block.FreshStake)
+
+	for _, tx := range data.RawSTx {
+		msgTx := txhelpers.MsgTxFromHex(tx.Hex)
+		if msgTx == nil {
+			log.Errorf("Unknown transaction %s", tx.Txid)
+			return nil
+		}
+		switch stake.DetermineTxType(msgTx) {
+		case stake.TxTypeSSGen:
+			stx := makeExplorerTxBasic(tx, msgTx)
+			stx.Fee = 0.0
+			stx.FeeRate = 0.0
+			votes = append(votes, stx)
+		case stake.TxTypeSStx:
+			stx := makeExplorerTxBasic(tx, msgTx)
+			tickets = append(tickets, stx)
+		case stake.TxTypeSSRtx:
+			stx := makeExplorerTxBasic(tx, msgTx)
+			revocations = append(revocations, stx)
+		}
+	}
+
+	txs := make([]*explorer.TxBasic, 0, block.Transactions)
+	for _, tx := range data.RawTx {
+		exptx := makeExplorerTxBasic(tx, txhelpers.MsgTxFromHex(tx.Hex))
+		for _, vin := range tx.Vin {
+			if vin.IsCoinBase() {
+				exptx.Fee, exptx.FeeRate = 0.0, 0.0
+			}
+		}
+		txs = append(txs, exptx)
+	}
+	block.Tx = txs
+	block.Votes = votes
+	block.Revs = revocations
+	block.Tickets = tickets
+
+	sortTx := func(txs []*explorer.TxBasic) {
+		sort.Slice(txs, func(i, j int) bool {
+			return txs[i].Total > txs[j].Total
+		})
+	}
+
+	sortTx(block.Tx)
+	sortTx(block.Votes)
+	sortTx(block.Revs)
+	sortTx(block.Tickets)
+
+	getTotalSent := func(txs []*explorer.TxBasic) (total float64) {
+		for _, tx := range txs {
+			total += tx.Total
+		}
+		return
+	}
+
+	getTotalSpent := func(txs []*explorer.TxBasic) (total float64) {
+		for _, tx := range txs {
+			total += tx.Fee.ToCoin()
```

Comment:
I assumed total sent is the sum of the input values and total spent is the sum of the fees

Reply:
The miner fees for a transaction are sum(inputs) - sum(outputs).  Perhaps it's confusing, but I refer to the sum of inputs as spent since it's how much you as the sender spend to make the tx happen, while sent is how much you are sending to the output address(es).  The difference is the fees that the miner gets.

The fees for a block could be just that sum `total += tx.Fee.ToCoin()` over all transactions `tx`.  Or it could be done the way you started it by computing sum of sum of inputs, but then it would need sum of sum of outputs for the difference to get mining fee.

Double check against the Insight explorer to be sure it makes sense.

Reply:
Changed MiningFee calculation to sum fees in over all `tx` except votes. I checked and the tx fees are the same on the Insight explorer.